xarray/test/__init__.py

```diff
@@ -215,6 +215,56 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         self.assertVariableAllClose(ar1, ar2, rtol=rtol, atol=atol)
         self.assertCoordinatesEqual(ar1, ar2)
 
+def assert_xarray_equal(a, b):
+    import xarray as xr
+    ___tracebackhide__ = True
+    assert type(a) == type(b)
+    if isinstance(a, xr.Dataset):
+        assert_xarray_equal(a.data_vars, b.data_vars)
+        assert_xarray_equal(a.coords, b.coords)
+    elif isinstance(a, (xr.Variable, xr.DataArray, xr.Coordinate)):
```

Comment:
`Coordinate` (actually `IndexVariable`, now) is a `Variable` subclass, so you don't need to call it out separately